DB/src/main/java/io/deephaven/db/v2/locations/parquet/local/DeephavenStylePartitionLayout.java

```diff
@@ -0,0 +1,86 @@
+package io.deephaven.db.v2.locations.parquet.local;
+
+import io.deephaven.db.v2.locations.TableDataException;
+import io.deephaven.db.v2.locations.TableLocationKey;
+import io.deephaven.db.v2.locations.local.PrivilegedFileAccessUtil;
+import io.deephaven.util.annotations.VisibleForTesting;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+/**
+ * {@link ParquetTableLocationScanner.LocationKeyFinder Parquet location finder} that will traverse a directory
+ * hierarchy laid out in Deephaven's "nested-partitioned" format, e.g.
+ * <pre>tableRootDirectory/internalPartitionValue/columnPartitionValue/tableName/table.parquet</pre>, producing
+ * {@link ParquetTableLocationKey}'s with two partitions, for keys {@value INTERNAL_PARTITION_KEY} and the specified
+ * {@code columnPartitionKey}.
+ */
+public final class DeephavenStylePartitionLayout implements ParquetTableLocationScanner.LocationKeyFinder {
```

Comment:
The name here feels a bit too commiting.

Perhaps we could call this style "Mortadela" ?  or "Pepito" ?  "Albaricoque" ?

More seriously, can we use some notable property of the partition layout for its name?